packages/wrangler/src/deployment-bundle/module-collection.ts

```diff
@@ -251,6 +269,51 @@ export function createModuleCollector(props: {
 								// it to `esbuild` to bundle it.
 								if (isJavaScriptModuleRule(rule)) return;
 
+								// Check if this file is possibly from an npm package
+								// and if so, validate the import against the package.json exports
+								// and resolve the file path to the correct file.
+								if (args.path.includes("/") && !args.path.startsWith(".")) {
+									// get npm package name from string, taking into account scoped packages
+									const packageName = extractPackageName(args.path);
+									if (!packageName) {
+										throw new Error(
+											`Unable to extract npm package name from ${args.path}`
+										);
+									}
+									const packageJsonPath = path.join(
+										args.resolveDir,
+										"node_modules",
+										packageName,
+										"package.json"
+									);
+									// Try and read the npm package's package.json
+									// and then resolve the import against the package's exports
+									// and then finally override filePath if we find a match.
+									try {
+										const packageJson = JSON.parse(
+											await readFile(packageJsonPath, "utf8")
+										);
+										const testResolved = resolveExports(
+											packageJson,
+											args.path.replace(`${packageName}/`, ""),
+											{
+												conditions: ["workerd", "worker", "browser"],
```

Comment:
Can you pull this out into a constant for sharing between here and https://github.com/cloudflare/workers-sdk/blob/f4ad634af86c49ade427af23e3853c656e30250a/packages/wrangler/src/deployment-bundle/bundle.ts#L313?

Reply:
Done, lemme know if you'd prefer this in another file